test/unit/plugins/vocablink.test.js

```diff
@@ -0,0 +1,42 @@
+const expect = require('expect');
+const parser = require('../../../src/cdoFlavoredParser');
+
+describe('vocablink', () => {
+  describe('render', () => {
+    it('cannot render vocablinks to html', () => {
+      const input = "[v some-word]";
+      const output = parser.sourceToHtml(input);
+      expect(output).toEqual("<p>[v some-word]</p>\n");
+    });
+  });
+
+  describe('redact', () => {
+    it('redacts vocablinks', () => {
+      const input = "[v some-word]";
+      const output = parser.sourceToRedacted(input);
+      expect(output).toEqual("[some-word][0]\n");
+    });
+
+    it('redacts vocablinks with word overrides', () => {
+      const input = "[v some-word][un-mot]";
+      const output = parser.sourceToRedacted(input);
+      expect(output).toEqual("[un-mot][0]\n");
+    });
+  });
+
+  describe('restore', () => {
+    it('can restore vocablinks back to markdown', () => {
+      const source = "[v some-word]";
+      const redacted = "[un-mot][0]"
+      const output = parser.sourceAndRedactedToMarkdown(source, redacted);
+      expect(output).toEqual("[v some-word][un-mot]\n");
+    });
+
+    it('can restore vocablinks with word overrides back to markdown', () => {
+      const source = "[v some-word][source-override]";
+      const redacted = "[redaction-override][0]"
```

Comment:
what is the scenario where we would need to override something that already has an override? probably not the translation scenario I asked about in the previous question, since that wouldn't explain the override in the original source.

Reply:
Mostly just to deal with the fact that the override might be in the original source.

A principle that I'm building all of this on is that the source text and restored text are both rendered by the exact same parser, so if we want to be able to have syntax that allows us to override some text with the translated version in the restored text we have to also accept that it's possible to use that syntax in source text